connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -160,20 +162,39 @@ public Connector createConnector(String listener) {
         return connector;
     }
 
-    public void start(HerderProvider herderProvider, Plugins plugins) {
```

Comment:
edit: by old code, I mean you can avoid some of the changes here (for example, the changes in RestServerTest.java https://github.com/apache/kafka/pull/6651/files#diff-15eb5e6f7e30dd7b8682eb0f7fad8a3a).

Reply:
I think it's better to remove the old `start(...)` entirely if it's not going to be used anywhere except in tests.
Preserving backwards compatibility isn't necessary here since this isn't public API. Normally I'd still want to err on the side of caution here, but a combined start-and-initialize method would encourage buggy code that's either susceptible to deadlock with REST extensions (when invoked before the herder is started) or breaks on port 0 (when invoked after the herder is started).

Reply:
alright, if the only impact is test cases, and that works only because they don't touch the herder, I'm ok with not doing this.